front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -0,0 +1,208 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { P } from "pino";
+
+import { getDataSources } from "@app/lib/api/data_sources";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { connectorsClient } from "@app/lib/connectors_client";
+import { DustAPI } from "@app/lib/dust_api";
+import { DataSource, Key, Provider } from "@app/lib/models";
+import { credentialsFromProviders } from "@app/lib/providers";
+import { Err, Ok, Result } from "@app/lib/result";
+import { new_id } from "@app/lib/utils";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { DataSourceType } from "@app/types/data_source";
+
+export type GetDataSourcesResponseBody = {
+  dataSources: Array<DataSourceType>;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<GetDataSourcesResponseBody>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    res.status(404).end();
+    return;
+  }
+
+  let dataSources = await getDataSources(auth);
+
+  switch (req.method) {
+    case "POST":
+      if (!auth.isBuilder()) {
+        return apiError(req, res, {
+          status_code: 401,
+          api_error: {
+            type: "app_auth_error",
+            message:
+              "Only the users that are `builders` for the current workspace can create a data source.",
+          },
+        });
+      }
+
+      const dataSourceName = "Slack (managed)";
+      const dataSoruceDescription = "Slack (managed)";
+      const dataSourceProdiverId = "openai";
+      const dataSourceModelId = "text-embedding-ada-002";
+      const dataSourceMaxChunkSize = 512;
```

Comment:
Did you find it performed better than the default 256 ?

Reply:
No, this was a bit random to be honest. I was assuming we'll adjust all these by the time we push to the customer.
I can revert to 256.